### PR TITLE
[EA-125] Add a "diagnosis" REST resource

### PR DESCRIPTION
**Pull request by BartlomiejRasztabiga**
Issue: https://issues.openmrs.org/browse/EA-125

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+@Controller
+@RequestMapping(value = "/rest/emrapi")
+public class DiagnosisController extends BaseRestController {
+
+	@Autowired
+	private DiagnosisService diagnosisService;
+
+	@RequestMapping(method = RequestMethod.GET, value = "/diagnosislist")
+	@ResponseBody
+	public List<Diagnosis> getDiagnosesList(@RequestParam("patient") Patient patient,
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java]: Should the fromDate be required?

File: omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+											@RequestParam("fromDate") Date fromDate) {
+		return diagnosisService.getDiagnoses(patient, fromDate);
+	}
+
+	@RequestMapping(method = RequestMethod.GET, value = "/uniquediagnosislist")
+	@ResponseBody
+	public List<Diagnosis> getUniqueDiagnosesList(@RequestParam("patient") Patient patient,
+			                                      @RequestParam("fromDate") Date fromDate) {
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java]: Should the fromDate be required?

File: omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java (hunk 1 above)
Comment (BartlomiejRasztabiga) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java]: It is required by getDiagnoses() method

File: omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java (hunk 2 above)
Comment (BartlomiejRasztabiga) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java]: It is required by getUniqueDiagnoses() method

File: omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Controller
+@RequestMapping(value = "/rest/emrapi")
+public class DiagnosisController extends BaseRestController {
+
+	@Autowired
+	private DiagnosisService diagnosisService;
+
+	@RequestMapping(method = RequestMethod.GET, value = "/diagnosislist")
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java]: Also from the talk thread, Darius seems to suggest "patientdiagnoses" for the url

**Comment by dkayiwa** [comment]
And by the way, have you tried to run the module with these changes and made this rest call?

File: omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java (hunk 1 above)
Comment (dkayiwa) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java]: The api allows it to be null.

File: omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java (hunk 2 above)
Comment (dkayiwa) [reply omod/src/main/java/org/openmrs/module/emrapi/web/controller/DiagnosisController.java]: The api allows it to be null.

**Comment by BartlomiejRasztabiga** [comment]
Do I have to download all these modules to run EMR API? 
image

**Comment by ivange94** [comment]
You can find the dependencies for emrapi here https://github.com/openmrs/openmrs-module-emrapi/blob/master/omod/src/main/resources/config.xml#L20-L28

And yes, you need to install all it's dependencies to run it. Note that, it's dependencies may also have dependencies of their own.

**Comment by BartlomiejRasztabiga** [comment]
And how can I know which version of these modules should I install? Always newest?

**Comment by ivange94** [comment]
The link above tells which module versions you need but just that it uses already defined variables. To know the value of these variables you can look at the properties section of the parent pom of a module. For example for this module you can find them here https://github.com/openmrs/openmrs-module-emrapi/blob/master/pom.xml#L40-L55

**Comment by BartlomiejRasztabiga** [comment]
Thanks a lot :)

**Pull request merged**